src/test/logical_query_plan/create_prepared_plan_node_test.cpp

```diff
@@ -44,6 +44,22 @@ TEST_F(CreatePreparedPlanNodeTest, Equals) {
   EXPECT_NE(*different_prepared_plan_node_b, *create_prepared_plan_node);
 }
 
+TEST_F(CreatePreparedPlanNodeTest, Hash) {
+  const auto deep_copy_node = create_prepared_plan_node->deep_copy();
+  EXPECT_EQ(create_prepared_plan_node->hash(), deep_copy_node->hash());
+
+  const auto different_prepared_plan_node_a = CreatePreparedPlanNode::make("some_prepared_plan2", prepared_plan);
+
+  const auto different_lqp = MockNode::make(MockNode::ColumnDefinitions({{DataType::Int, "b"}}));
+  const auto different_prepared_plan =
+      std::make_shared<PreparedPlan>(different_lqp, std::vector<ParameterID>{ParameterID{1}});
+  const auto different_prepared_plan_node_b =
+      CreatePreparedPlanNode::make("some_prepared_plan", different_prepared_plan);
+
+  EXPECT_NE(different_prepared_plan_node_a->hash(), create_prepared_plan_node->hash());
+  EXPECT_EQ(different_prepared_plan_node_b->hash(), create_prepared_plan_node->hash());
```

Comment:
These tests look very much like the equals tests. Wouldn't it be shorter to combine the two?